### PR TITLE
Use poetry

### PR DESCRIPTION
**Pull request by frazane**
This PR introduces poetry for dependency management and packaging. These are the most important changes:

- added a new `pyproject.toml` file with poetry-based configuration
- `poetry.lock` file is automatically created 
- removed unnecessary files: `MANIFEST`, `setup.py` and `.readthedocs.yml`
- miscellaneous changes to comply with lint rules

### CODE REVIEW

**Review by ghiggi — COMMENTED** [review]
I just went through your edits and added some notes. I never used poetry so I let you test it works ;=)

File: pyproject.toml [hunk 1]
```diff
@@ -1,147 +1,85 @@
...
-line-length = 100
-# skip-string-normalization = true
-target-version = [
-    "py37",
-    "py38",
-    "py39",
-    "py310",
-    "py311",
-]
+line-length = 88
```
[the remark this answers is not in the document]
Comment (frazane) [reply pyproject.toml:42]: PEP8 would be 79, black's 88 is a good compromise IMO 🙂

File: MANIFEST.in [hunk 2]
```diff
@@ -1,4 +0,0 @@
-prune .github
-prune dev
```
[the remark this answers is not in the document]
Comment (frazane) [reply MANIFEST.in:2]: By default poetry only considers files in the source directory. Anything extra needs to/can be specified in the `pyproject.toml` (see https://python-poetry.org/docs/pyproject/#include-and-exclude). Short story: MANIFEST files are obsolete

File: pyproject.toml [hunk 3]
```diff
@@ -1,147 +1,85 @@
...
-Tracker = "https://github.com/ghiggi/xverif/issues"
-Documentation = "https://xverif.readthedocs.io"
-changelog = "https://github.com/ghiggi/xverif/CHANGELOG.md"
 
-[tool.setuptools_scm]
-write_to = "xverif/_version.py"
+[tool.poetry.group.dev.dependencies]
+pytest = "^7.3.1"
+ruff = "^0.0.265"
+black = "^23.3.0"
```
[the remark this answers is not in the document]
Comment (frazane) [reply pyproject.toml:23]: It's probably worth including it but I'd leave it for the next PR (where we introduce notebooks and the new docs).

File: pyproject.toml [hunk 4]
```diff
@@ -1,147 +1,85 @@
...
+isort = "^5.12.0"
```
[the remark this answers is not in the document]
Comment (frazane) [reply pyproject.toml:24]: You're right!

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -1,23 +1,39 @@
...
+  autofix_prs: false
+  skip: ["ruff"]
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v4.4.0
     hooks:
       - id: trailing-whitespace
       - id: end-of-file-fixer
 
-  - repo: https://github.com/charliermarsh/ruff-pre-commit
```
[the remark this answers is not in the document]
Comment (frazane) [reply .pre-commit-config.yaml:8]: We run it locally (see below), and then it's a separate github workflow for PR actions.

**Pull request merged**